### PR TITLE
feat: use defopt instead of click for parsing

### PR DESCRIPTION
**Pull request by nh13**
Note: defopt is very opinionated on the single and long flag namings, so there is some backwards incompatibility (e.g. `--file-name` versus `--file_name`).

### CODE REVIEW

File: plannotate/pLannotate.py [hunk 1]
```diff
@@ -77,68 +77,79 @@ def main_setupdb():
...
+
+def main(argv: List[str] = sys.argv[1:]) -> None:
+    tools: List[Callable] = [
+        batch,
+        setupdb,
+        streamlit,
+        yaml,
+    ]
+    if len(argv) != 0 and all(arg not in argv for arg in ["-h", "--help"]):
+        print("Running command: client-tools " + " ".join(argv))
```
Comment (nh13) [inline plannotate/pLannotate.py:145]: could use logging instead of print here.

**Comment by jeffreybarrick** [comment]
I'm leaving comments/decision/testing of this to @mmcguffi 

My only comments are that I prefer the hyphen format for multiword options and that currently if I type `plannotate batch` I get an error versus the help message that I expected for running with no options.

**Comment by jeffreybarrick** [comment]
My comment was about the current code not the updated PR code... glad to see it does print the help with the updated code.

**Comment by mmcguffi** [comment]
`defopt` seems interesting and I like how elegant it appears to be. I'll dig into it a little more and decide then -- thanks for writing up the PR at a minimum.

Just an FYI I'm currently  in the process of moving across the country and starting a new job, so I won't be able to test/address these immediately, but I definitely will once I am settled.

**Comment by nh13** [comment]
No urgency here. This is all just trying to give back as I find things.

**Comment by mmcguffi** [comment]
I very much appreciate the effort on this @nh13, but I am going to stick with `click` for now -- maybe this is a foolish, sisyphean choice since `click` has burned me so many times, but I guess we'll see...

At some point I intend to refactor and add type hints/docstrings throughout, so maybe I'll re-assess `defopt` then

Thanks again for the help and labor!

**Pull request closed without merging**